source/common/rate_limiter_impl.h

```diff
@@ -157,4 +177,44 @@ class FilteringRateLimiterImpl : public ForwardingRateLimiterImpl,
   const RateLimiterFilter filter_;
 };
 
+/**
+ * Takes a probabilistic approach to suppressing an arbitrary wrapper rate limiter.
+ */
+class GraduallyOpeningRateLimiterFilter : public FilteringRateLimiterImpl {
+public:
+  GraduallyOpeningRateLimiterFilter(const std::chrono::nanoseconds ramp_time,
+                                    DiscreteNumericDistributionSamplerPtr&& provider,
+                                    RateLimiterPtr&& rate_limiter);
+
+private:
+  DiscreteNumericDistributionSamplerPtr provider_;
+  const std::chrono::nanoseconds ramp_time_;
+};
+
+/**
+ * Thin wrapper around absl::zipf_distribution that will pull zeroes and ones from the distribution
+ * with the intent to probabilistically suppress the wrapped rate limiter.
+ * This may need further consideration, because it will shoot holes in the pacing, lowering the
+ * actual achieved frequency.
+ */
+class ZipfRateLimiterImpl : public FilteringRateLimiterImpl {
```

Comment:
Yes, let's move this into its own PR. I'm exploring an initial take on this here to figure out how we  this could fit zipf distributions into our current model / get a sense of how we can land this. @htuch came up with the idea for this feature and I think it would be a really cool to have.

Having said that, I feel the zipf stuff is kind of on a different trajectory compared to the other rate limiters in here. 

Initially I planned to only do the ramping ones as that can be used directly to get us a better warm-up phase, but maybe I got a little (over-)enthusiastic ;-)

Reply:
Moved the zipf stuff over to #251

Reply:
Thank you for moving it over to another PR. Let's continue the discussion over there, I am interested to hear more about the Zipf use case and motivation.